lib/bloc/invoice/invoice_bloc.dart

```diff
@@ -113,11 +113,18 @@ class InvoiceBloc {
       }
       return s;
     })
-    .asyncMap((paymentRequest) async { 
+    .asyncMap((paymentRequest) async {
+      // add stream event before processing and decoding
+      _receivedInvoicesController.add(
+          new PaymentRequestModel(null, paymentRequest));
       //filter out our own payment requests
       var invoice;
       try {
-        invoice = await breezLib.getRelatedInvoice(paymentRequest);
+        invoice = await breezLib.getRelatedInvoice(paymentRequest).then((invoice) {
+          if(invoice == null){
+            _receivedInvoicesController.addError("Invalid Invoice");
```

Comment:
The related invoice is a mechanism to check if this is our own invoice. In that case we shouldn't do anything.
Since here an error is added , I think right now it will show a flushbar error for such invoices.